tests/system/datastore/test_v3io.py

```diff
@@ -86,73 +86,104 @@ def _setup_df_dir(self, first_file_path, second_file_path, file_extension):
-            object_path = urlparse(object_url).path
-            data_item.store._upload(
-                object_path, tempfile_1_path, max_chunk_size=100 * 1024
-            )
-
-            self._logger.debug("Downloading the object")
-            data_item.download(tempfile_2_path)
-
-            cmp_process = subprocess.Popen(cmp_command, stdout=subprocess.PIPE)
-            stdout, stderr = cmp_process.communicate()
-            assert (
-                cmp_process.returncode == 0
-            ), f"stdout = {stdout}, stderr={stderr}, returncode={cmp_process.returncode}"
-
-        finally:
-            # cleanup (local files are cleaned by the TempDir destructor)
-            data_item.delete()
+        self._logger.debug(
+            f"test_v3io_large_object_upload - finished to write locally in {time.time() - first_start_time} seconds"
+        )
+        self._logger.debug(
+            "Exercising the DataItem upload flow",
+            tempfile_1_path=tempfile_1_path,
+            tempfile_2_path=tempfile_2_path,
+        )
+        start_time = time.time()
+        data_item.upload(tempfile_1_path)
+        self._logger.debug(
+            f"test_v3io_large_object_upload - finished to upload in {time.time() - start_time} seconds"
+        )
+        start_time = time.time()
+        data_item.download(tempfile_2_path)
+        self._logger.debug(
+            f"test_v3io_large_object_upload - finished to download locally in {time.time() - start_time} seconds"
+        )
+        start_time = time.time()
+        cmp_process = subprocess.Popen(cmp_command, stdout=subprocess.PIPE)
+        stdout, stderr = cmp_process.communicate()
+        assert (
+            cmp_process.returncode == 0
+        ), f"stdout = {stdout}, stderr={stderr}, returncode={cmp_process.returncode}"
+        self._logger.debug(
+            f"test_v3io_large_object_upload - finished cmp 1 in {time.time() - start_time} seconds"
+        )
+        # Do the test again, this time exercising the v3io datastore _upload() loop
+        self._logger.debug("Exercising the v3io _upload() loop")
+        os.remove(tempfile_2_path)
+        object_path = urlparse(object_url).path
+        start_time = time.time()
+        data_item.store._upload(object_path, tempfile_1_path, max_chunk_size=100 * 1024)
+        self._logger.debug(
+            f"test_v3io_large_object_upload - finished to upload with store directly in"
+            f" {time.time() - start_time} seconds"
+        )
+        self._logger.debug("Downloading the object")
+        start_time = time.time()
+        data_item.download(tempfile_2_path)
+        self._logger.debug(
+            f"test_v3io_large_object_upload - finished to download in the second time in"
+            f" {time.time() - start_time} seconds"
+        )
+        start_time = time.time()
+        cmp_process = subprocess.Popen(cmp_command, stdout=subprocess.PIPE)
+        stdout, stderr = cmp_process.communicate()
+        assert (
```

Comment:
we might also add an assertion that the downloaded data_item is the same as the original.